src/TextInput.js

```diff
@@ -30,7 +30,7 @@ const TextInput = ({icon, type, className, width, theme, ...rest}) => {
   const inputClasses = classnames(icon ? 'input-icon' : 'input-no-icon')
   const hasIcon = !!icon
   return (
-    <Wrapper width={width} className={wrapperClasses} theme={theme}>
```

Comment:
@emplums I was confused about this. Adding this `width` attribute here results in markup that has an actual `width` attribute on the `span` element, which is not valid DOM. This was on my list of things to revisit.

Reply:
@T-Hugs That's a known issue with styled-components, it doesn't seem to reliably remove the style props from the attributes that are eventually passed down to the DOM. It looks ugly, but don't make too much of difference, so I wouldn't worry about it! But we do need to move the width prop back to avoid the breaking change. As it is currently in this branch, the width doesn't get applied at all.

Reply:
I also noticed that the `COMMON` props that should be applied to the `Wrapper` component don't work because we're not spreading them onto that component. Related - I'm not sure if there was a reason why `width` was originally set on the `Wrapper` to begin with, but if it's not necessary to set on the wrapper we could move the `width` and `COMMON` style functions directly to the `Input` component and then we don't need to pick & choose which props should be forwarded to `Wrapper` and which should be forwarded to `Input`, we can just spread all the props onto `Input` and take care of styling there. 

![image](https://user-images.githubusercontent.com/8960591/70528726-2bd58000-1b4f-11ea-9d2f-074935a891dc.png)

I'm starting to be a little worried that `TextInput` is getting a bit too big by trying to accommodate the `FilterInput` use case - for example, the `Wrapper` component was added to make it work with `FilterInput` and I'm not sure how the changes above in the `Wrapper` component will affect `FilterInput`  - to me that's a signal that it should probably be separated.

Are there a lot of use cases in Primer that require an icon and the special wrapper styling other than the FilterInput? If not, we should probably have a "from scratch" input in `FilterInput` and not try to build off of `TextInput` and thus expand it's API surface area.

Reply:
If you're cool with that, I'm happy to make that change in this branch!

Reply:
> Are there a lot of use cases in Primer that require an icon and the special wrapper styling other than the FilterInput? If not, we should probably have a "from scratch" input in FilterInput and not try to build off of TextInput and thus expand it's API surface area.

My gut feeling is that icons on inputs are common enough to be warrented here. Looking around GitHub UI, I found 3 different icons that accompany inputs.

Reply:
Regarding moving common to the input, I've done that in a new commit and it seems to work great! Thanks for the suggestion.